plugin/core/sessions.py

```diff
@@ -429,16 +448,22 @@ class Session(Client):
 
     def __init__(self, manager: Manager, logger: Logger, workspace_folders: List[WorkspaceFolder],
                  config: ClientConfig, plugin_class: Optional[Type[AbstractPlugin]]) -> None:
+        super().__init__(logger)
         self.config = config
         self.manager = weakref.ref(manager)
         self.window = manager.window()
         self.state = ClientStates.STARTING
         self.capabilities = DottedDict()
+        self.exiting = False
+        self.views_opened = 0
```

Comment:
Possibly relevant https://stackoverflow.com/questions/52693294/is-it-safe-to-give-a-python-weakset-to-a-list-constructor

Reply:
> Can't we just check the size of self._session_views directly in instead of introducing this count?

No:
```
t0: last view close, debounced check after 3 seconds
t0 + 1s: open another view
t0 + 2s: close that view
t0 + 3s: debounced check is done, all views have been closed so we end
```
This would mean the session ends 1 second after apparently closing the last view. Using a counter prevents this.

Reply:
I wonder why `SessionView` handles that then. Couldn't `Session` handle debouncing when called from `SessionView` through `unregister_session_view_async`?

Reply:
That’s probably a better idea, thanks.

Reply:
https://github.com/sublimelsp/LSP/pull/1069/commits/b20518d00d0544aaf7b28ef927fe36c1cb785921